types/address.go

```diff
@@ -168,67 +174,39 @@ func VerifyAddressFormat(bz []byte) error {
 	if verifier != nil {
 		return verifier(bz)
 	}
-	if len(bz) != AddrLen {
-		return fmt.Errorf("incorrect address length (expected: %d, actual: %d)", AddrLen, len(bz))
+	if len(bz) != BytesAddrLen {
+		if len(bz) == 0 {
+			return errors.New("empty address string is not allowed")
+		}
+		return fmt.Errorf("incorrect address length (expected: %d, actual: %d)", BytesAddrLen, len(bz))
 	}
 	return nil
 }
 
-// AccAddressFromBech32 creates an AccAddress from a Bech32 string.
-func AccAddressFromBech32(bech32Addr string) (AccAddress, error) {
-	addr, ok := bech32Cache.GetAddr(bech32Addr)
-	if ok {
-		return addr, nil
-	}
-
-	if len(strings.TrimSpace(bech32Addr)) == 0 {
-		return AccAddress{}, errors.New("empty address string is not allowed")
-	}
-
-	bech32PrefixAccAddr := GetConfig().GetBech32AccountAddrPrefix()
-
-	bz, err := GetFromBech32(bech32Addr, bech32PrefixAccAddr)
-	if err != nil {
-		return nil, err
-	}
-
-	err = VerifyAddressFormat(bz)
-	if err != nil {
-		return nil, err
-	}
-	bech32Cache.Set(bech32Addr, bz)
-	return bz, nil
-}
-
 // Returns boolean for whether two AccAddresses are Equal
 func (aa AccAddress) Equals(aa2 Address) bool {
 	if aa.Empty() && aa2.Empty() {
 		return true
 	}
 
-	return bytes.Equal(aa.Bytes(), aa2.Bytes())
+	return strings.EqualFold(aa.String(), aa2.String())
 }
 
 // Returns boolean for whether an AccAddress is empty
 func (aa AccAddress) Empty() bool {
-	if aa == nil {
-		return true
-	}
-
-	aa2 := AccAddress{}
-	return bytes.Equal(aa.Bytes(), aa2.Bytes())
+	return len(string(aa)) == 0
```

Comment:
It's false. Empty byte generates an address `link1g79a9a`. It is no problem to judge that this address(`link1g79a9a`) is not empty, given that the use of `Empty()` is intended for a prior examination so that the operation of `AccAddress` is not error-proof.

Reply:
It's reasonable.
Then, How about using `Empty()` for address checking also?
e.g.) addr == "" -> addr.Empty()
https://github.com/line/lfb-sdk/pull/298/files#diff-b618e5d1a7bdaf97e1e5ddc6b10a2c8a907e0a1d2be54a2eee320320e69a5908L79

Reply:
Okay, I'll apply it.

Reply:
🙇‍♂️